pkg/kubectl/cmd/new.go

```diff
@@ -0,0 +1,225 @@
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/spf13/cobra"
+	"io"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+const (
+	new_long = `Create a new object of the given type using a template.
+
+The new command pulls down an for the given object type, allows you
+to edit the filled in example, and then submits it for creation.  The
+editing functionality functions similarly to the 'edit' command -- you
+can control the editor used, as well as the format and version, using
+the same flags and environment variables.
+
+By default, before submission your new object will be validated.  If an
+invalid object is encountered, you will be returned to the editor with the
+invalid part indicated with comments.  Use the '--validate=false' in order
+to skip local validation.
+
+You can specify namespace, name, and labels (in 'key1=value1,key2=value2' form)
+from the command line using the '--namespace', '--name', and '--labels' flags.
+
+You can use the '--dry-run' flag to just output the example to standard out
+without launching an editor or submitting the object for creation.
+`
+
+	new_example = `# Create a new pod named 'website':
+  $ kubectl create pod --name website --labels end=front,app=site
+
+  # Don't validate the object before creation
+  $ kubectl create namespace --validate=false --name my-ns
+
+  # Don't edit or create a new object -- just output the example
+  $ kubectl create namespace --name my-ns --dry-run`
+)
+
+func NewCmdNew(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "new TYPE [--name NAME] [--labels key1=val,key2=val]",
+		Short:   "Create a new object of the given type by filling in a template",
+		Long:    new_long,
+		Example: new_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunNew(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: 'yaml' or 'json'.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	cmd.PersistentFlags().String("name", "", "The name of the new object")
+	cmd.PersistentFlags().String("labels", "", "A comma-separated list of labels in key=value form to apply to the created object")
+	cmd.Flags().Bool("dry-run", false, "If true, only print the example object, without editing or creating it.")
+
+	cmdutil.AddValidateFlags(cmd)
+
+	return cmd
+}
+
+func initExample(cmdNamespace string, resource string, f *cmdutil.Factory, cmd *cobra.Command) (runtime.Object, error) {
+	mapper, _ := f.Object()
+	version, kind, err := mapper.VersionAndKindForResource(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	exFetcher := cmdutil.NewStaticExampleFetcher(api.Scheme)
```

Comment:
This interface needs to come from the `Factory` as a new method on it (returns the interface, which you then invoke).